v2/pkg/archive/image-blob-gatherer.go

```diff
@@ -60,39 +61,43 @@ func (o *ImageBlobGatherer) GatherBlobs(ctx context.Context, imgRef string) (blo
 	}
 	blobs[digest.String()] = ""
 
-	if manifest.MIMETypeIsMultiImage(mime) {
-		manifestList, err := manifest.ListFromBlob(manifestBytes, mime)
-		if err != nil {
-			return nil, err
-		}
-		instances := manifestList.Instances()
-		for _, digest := range instances {
-			blobs[digest.String()] = ""
-			singleArchManifest, singleArchMime, err := img.GetManifest(ctx, &digest)
+	// it seems this code never gets called
+	// the manifest list gets moved to sha256 hash with .manifest.json
+	// i.e ad59206271872ca50d4b202c8298018c53bfaefd03b64ee036ad5362fc0214ce.manifest.json
+	/*
```

Comment:
I’m afraid I lack context. Assuming this function is intended to be _generic_ over all valid inputs of `alltransports.ParseImageName(imgRef)`, it doesn’t make sense to talk about paths like that comment does.

If that refers to the `dir:` transport, no, the manifest list would be the top-level `${path}/manifest.json`, and the per-platform individual manifests would be `${path}/${digest}.manifest.json`. The top-level manifest list is never stored in `${digest}.manifest.json`.

---

As a purely abstract matter, I would definitely expect a mirroring tool to be able to mirror multi-platform images as a whole; if nothing else, because the OpenShift release image uses digest references to refer to the other product images, so no modifications at all are acceptable for those images. It might also have some functionality to not do that (either to mirror only a single arch, turning a multi-platform image into a single-platform one, or to only copy some of the per-platform images).

But I also never read the `oc-mirror` documentation; you know what your software is designed to do.

_If_ this function is intended to handle multi-platform images, the codebase _must_ have the “list all platforms, iterate over each per-platform manifest” code _somewhere_. That has nothing to do with schema2/OCI formats, just with the ability to deal with multi-platform images at all. `manifest.FromBlob` for a multi-platform manifest list outright fails.